Sources/SnapshotTesting/Strategy/View.swift

```diff
@@ -0,0 +1,213 @@
+#if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
+import SceneKit
+import SpriteKit
+import WebKit
+import WKSnapshotConfigurationShim
+
+#if os(macOS)
+extension Strategy where A == NSView, B == NSImage {
+  public static var view: Strategy {
+    return .view(precision: 1)
+  }
+
+  public static func view(precision: Float) -> Strategy {
+    return Strategy<NSImage, NSImage>.image(precision: precision).asyncPullback { view in
+      return view.snapshot ?? Async { callback in
+        addImagesForRenderedViews(view).sequence().run { views in
```

Comment:
`addImagesForRenderedViews` returns `[Async<View>]`, a bunch of image view renderings that have been inserted in the view hierarchy so we can remove them later. We call `sequence` to transform it into `Async<[View]>`, run it so it calls back when all of the image views have been inserted, and finally we can snapshot the view before removing the image views.